### ISSUE TITLE
Multi-Vector Timeout

### ISSUE THREAD

**Issue by aamir-s18** [vespa-engine/vespa#26730]
**Describe the bug**
TLDR: Getting timeout when adding around 7k vectors to multi-vector. However, after some time the document occurs when calling vespa visit.

Hey, I was following the [multi-vector](https://github.com/vespa-engine/sample-apps/tree/master/multi-vector-indexing) example. Based on it, I made some minor adjustments (find the config below). When I add a small number of paragraphs, it works without any issues. Still, if I try to add larger documents (in this example 7k paragraphs), I get the error "context deadline exceeded (Client.Timeout exceeded while awaiting headers)" with the vespa CLI. Interestingly the document occurs when calling vespa visit. I assume the post is successful, but Vespa hasn't proceeded yet with calculating the embeddings.

**To Reproduce**
1. Used sample application [multi-vector-indexing](https://github.com/vespa-engine/sample-apps/tree/master/multi-vector-indexing)
2. Used the following schemas:
```
schema test {

  document test {

    field file_id type string {
      indexing: attribute | summary
    }

    field filename type string {
      indexing: index | summary
      index: enable-bm25
    }

    field author type string {
      indexing: index | summary
      index: enable-bm25
    }

    field paragraphs type array<string> {
      indexing: index | summary
      index: enable-bm25
      bolding: on 
    }
  }

  field paragraph_embeddings type tensor<float>(p{},x[384]) {
    indexing: input paragraphs | embed | index | attribute
    attribute {
      distance-metric: angular
    }
  }

  fieldset default {
    fields: filename, author, paragraphs 
  }

  rank-profile semantic inherits default {
    inputs {
      query(q) tensor<float>(x[384])
    }
    first-phase {
      expression: cos(distance(field,paragraph_embeddings))
    }
    match-features {
      closest(paragraph_embeddings)
    }
  }

  rank-profile bm25 {
    first-phase {
      expression: 2*bm25(filename) + bm25(paragraphs) 
    }
  }

  rank-profile hybrid inherits semantic {
    function avg_paragraph_similarity() {
      expression {
        reduce(
          sum(l2_normalize(query(q),x) * l2_normalize(attribute(paragraph_embeddings),x),x),
          avg,
          p
        )
      }
    }
    function max_paragraph_similarity() {
      expression {
        reduce(
          sum(l2_normalize(query(q),x) * l2_normalize(attribute(paragraph_embeddings),x),x),
          max,
          p
        )
      }
    }

    function all_paragraph_similarities() {
      expression {
        sum(l2_normalize(query(q),x) * l2_normalize(attribute(paragraph_embeddings),x),x)
      }
    }

    first-phase {
      expression: cos(distance(field,paragraph_embeddings)) 
    }

    second-phase {
      expression {
        firstPhase + avg_paragraph_similarity() + log( bm25(filename) + bm25(paragraphs) + bm25(author))
      }
    }

    match-features {
      firstPhase
      closest(paragraph_embeddings)
      bm25(filename)
      bm25(paragraphs)
      avg_paragraph_similarity
      max_paragraph_similarity
      all_paragraph_similarities
    }
  }

  document-summary minimal {
    summary file_id type string {}
    summary filename type string {}
  }
}
```
3. Trying to feed the data using the vespa cli. The error still occurs when I use pyvespa or curl.

**Expected behavior**
It should ingest the data and return the path to the document. Or something like an ID to check how far the proceeding is.

**Logs**
Logs from vespa.log
```
[2023-04-13 10:15:34.903] INFO    container        stdout       [248.695s][info][gc] GC(40) Pause Young (Normal) (G1 Evacuation Pause) 1235M->387M(1536M) 4.234ms
[2023-04-13 10:15:37.331] INFO    container        stdout       [251.111s][info][gc] GC(41) Pause Young (Normal) (G1 Evacuation Pause) 1247M->387M(1536M) 7.598ms
[2023-04-13 10:15:40.097] INFO    container        stdout       [253.890s][info][gc] GC(42) Pause Young (Normal) (G1 Evacuation Pause) 1269M->387M(1536M) 9.072ms
[2023-04-13 10:15:42.828] INFO    container        stdout       [256.620s][info][gc] GC(43) Pause Young (Normal) (G1 Evacuation Pause) 1287M->388M(1536M) 5.103ms
[2023-04-13 10:15:45.471] INFO    container        stdout       [259.263s][info][gc] GC(44) Pause Young (Normal) (G1 Evacuation Pause) 1300M->389M(1536M) 20.452ms
[2023-04-13 10:15:48.097] INFO    container        stdout       [261.889s][info][gc] GC(45) Pause Young (Normal) (G1 Evacuation Pause) 1301M->390M(1536M) 8.291ms
[2023-04-13 10:15:50.619] INFO    container        stdout       [264.412s][info][gc] GC(46) Pause Young (Normal) (G1 Evacuation Pause) 1301M->391M(1536M) 5.957ms
[2023-04-13 10:15:53.230] INFO    container        stdout       [267.023s][info][gc] GC(47) Pause Young (Normal) (G1 Evacuation Pause) 1301M->392M(1536M) 4.805ms
[2023-04-13 10:15:55.908] INFO    container        stdout       [269.700s][info][gc] GC(48) Pause Young (Normal) (G1 Evacuation Pause) 1301M->392M(1536M) 9.215ms
[2023-04-13 10:15:58.544] INFO    container        stdout       [272.337s][info][gc] GC(49) Pause Young (Normal) (G1 Evacuation Pause) 1301M->393M(1536M) 6.254ms
[2023-04-13 10:16:01.243] INFO    container        stdout       [275.035s][info][gc] GC(50) Pause Young (Normal) (G1 Evacuation Pause) 1301M->394M(1536M) 12.118ms
[2023-04-13 10:16:03.892] INFO    container        stdout       [277.684s][info][gc] GC(51) Pause Young (Normal) (G1 Evacuation Pause) 1300M->396M(1536M) 9.562ms
[2023-04-13 10:16:06.509] INFO    container        stdout       [280.290s][info][gc] GC(52) Pause Young (Normal) (G1 Evacuation Pause) 1301M->396M(1536M) 6.651ms
[2023-04-13 10:16:09.150] INFO    container        stdout       [282.942s][info][gc] GC(53) Pause Young (Normal) (G1 Evacuation Pause) 1301M->397M(1536M) 7.331ms
[2023-04-13 10:16:11.838] INFO    container        stdout       [285.631s][info][gc] GC(54) Pause Young (Normal) (G1 Evacuation Pause) 1301M->397M(1536M) 4.964ms
[2023-04-13 10:16:14.471] INFO    container        stdout       [288.264s][info][gc] GC(55) Pause Young (Normal) (G1 Evacuation Pause) 1301M->399M(1536M) 5.904ms
[2023-04-13 10:16:17.165] INFO    container        stdout       [290.957s][info][gc] GC(56) Pause Young (Normal) (G1 Evacuation Pause) 1306M->399M(1536M) 5.979ms
[2023-04-13 10:16:19.915] INFO    container        stdout       [293.707s][info][gc] GC(57) Pause Young (Normal) (G1 Evacuation Pause) 1307M->400M(1536M) 4.773ms
[2023-04-13 10:16:22.598] INFO    container        stdout       [296.390s][info][gc] GC(58) Pause Young (Normal) (G1 Evacuation Pause) 1308M->400M(1536M) 5.313ms
[2023-04-13 10:16:25.319] INFO    container        stdout       [299.112s][info][gc] GC(59) Pause Young (Normal) (G1 Evacuation Pause) 1308M->401M(1536M) 5.179ms
[2023-04-13 10:16:28.005] INFO    container        stdout       [301.798s][info][gc] GC(60) Pause Young (Normal) (G1 Evacuation Pause) 1309M->403M(1536M) 5.909ms
[2023-04-13 10:16:30.731] INFO    container        stdout       [304.511s][info][gc] GC(61) Pause Young (Normal) (G1 Evacuation Pause) 1310M->403M(1536M) 4.918ms
[2023-04-13 10:16:33.326] INFO    container        stdout       [307.118s][info][gc] GC(62) Pause Young (Normal) (G1 Evacuation Pause) 1311M->403M(1536M) 5.865ms
[2023-04-13 10:16:36.056] INFO    container        stdout       [309.848s][info][gc] GC(63) Pause Young (Normal) (G1 Evacuation Pause) 1311M->405M(1536M) 4.676ms
[2023-04-13 10:16:38.676] INFO    container        stdout       [312.457s][info][gc] GC(64) Pause Young (Normal) (G1 Evacuation Pause) 1313M->405M(1536M) 6.417ms
[2023-04-13 10:16:41.244] INFO    container        stdout       [315.036s][info][gc] GC(65) Pause Young (Normal) (G1 Evacuation Pause) 1313M->406M(1536M) 7.433ms
[2023-04-13 10:16:43.900] INFO    container        stdout       [317.692s][info][gc] GC(66) Pause Young (Normal) (G1 Evacuation Pause) 1314M->407M(1536M) 6.021ms
[2023-04-13 10:16:46.542] INFO    container        stdout       [320.334s][info][gc] GC(67) Pause Young (Normal) (G1 Evacuation Pause) 1315M->408M(1536M) 5.081ms
```

**Environment (please complete the following information):**
 - OS: Debian
 - Infrastructure: self-hosted, Docker
 - Version: 8.148.14
 - 16 GB RAM, 6 CPU cores

**Vespa version**
8.148.14

**Comment by jobergum** [comment]
Could you include the *exact* details on the curl and the pyvespa exception? With 7K paragraphs embedded by invoking the model, you are likely running into a client-side request timeout. 

See https://everything.curl.dev/usingcurl/timeouts, or https://pyvespa.readthedocs.io/en/latest/reference-api.html 

```
vespa document -T 300 
```

Will update the doc accordingly.

**Comment by baldersheim** [comment]
Too old to be **soon**

**Comment by jobergum** [comment]
New section added to https://docs.vespa.ai/en/embedding.html#embedder-performance
